sky/packages/sky/lib/rendering/object.dart

```diff
@@ -439,6 +538,12 @@ abstract class RenderObject extends AbstractNode implements HitTestTarget {
     });
     _nodesNeedingLayout.add(this);
   }
+
+  /// Update the layout information for all dirty render objects
+  ///
+  /// This function is one of the core stages of the rendering pipeline. Layout
+  /// information is cleaned prior to painting so that render objects will
+  /// appear on screen in their up-to-date locations.
```

Comment:
Give a pointer to the code that calls this, for people trying to track down how it all works.